### PR TITLE
Fix NullReferenceException in RepeatedAssertionAnalyzer

### LINKED ISSUE #1328: 'FakeItEasy.Analyzer.RepeatedAssertionAnalyzer' threw an exception of type 'System.NullReferenceException'
FIE v4.5.0
The `CS8032` disappeared after restarting VS 2017 already upgraded to 4.5.0 (from 4.4.0)

```
Severity	Code	Description	Project	File	Line	Suppression State	Detail Description
Warning	AD0001	Analyzer 'FakeItEasy.Analyzer.RepeatedAssertionAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.	XXX.Tests	D:\Source\XXX.Tests\CSC	1	Active
Warning	CS8032	An instance of analyzer FakeItEasy.Analyzer.RepeatedAssertionAnalyzer cannot be created from C:\Users\XXX\.nuget\packages\fakeiteasy.analyzer.csharp\4.5.0\analyzers\dotnet\cs\FakeItEasy.Analyzer.CSharp.dll: Could not load type 'FakeItEasy.Analyzer.RepeatedAssertionAnalyzer' from assembly 'FakeItEasy.Analyzer.CSharp, Version=4.0.0.0, Culture=neutral, PublicKeyToken=eff28e2146d5fd2c'..	XXX.Tests		1	Active	System.TypeLoadException: Could not load type 'FakeItEasy.Analyzer.RepeatedAssertionAnalyzer' from assembly 'FakeItEasy.Analyzer.CSharp, Version=4.0.0.0, Culture=neutral, PublicKeyToken=eff28e2146d5fd2c'.
   at System.Reflection.RuntimeAssembly.GetType(RuntimeAssembly assembly, String name, Boolean throwOnError, Boolean ignoreCase, ObjectHandleOnStack type)
   at System.Reflection.RuntimeAssembly.GetType(String name, Boolean throwOnError, Boolean ignoreCase)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerFileReference.Extensions`1.GetAnalyzersForTypeNames(Assembly analyzerAssembly, IEnumerable`1 analyzerTypeNames, Boolean& reportedError)
-----


```

### PR DESCRIPTION
**Pull request by thomaslevesque**
Fixes #1328

### CODE REVIEW

File: tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/DiagnosticVerifier.Helper.cs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+using System.Runtime.ExceptionServices;
```
Comment (blairconrad) [inline tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/DiagnosticVerifier.Helper.cs]: inside the namespace

Comment (thomaslevesque) [reply tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/DiagnosticVerifier.Helper.cs]: oops

Comment (thomaslevesque) [reply tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/DiagnosticVerifier.Helper.cs]: Fixed (I made a fixup and squashed immediately)

**Comment by thomaslevesque** [comment]
> In addition to the using nit, do you want to add a test for typeless expressions in VB?

OK, but I'm not sure VB has the same behavior...

**Comment by blairconrad** [comment]
> OK, but I'm not sure VB has the same behavior...

It's not a huge deal. If `NameOf` doesn't do the same thing, don't spend all day on it or anything. :wink:

**Comment by thomaslevesque** [comment]
The bug doesn't happen in VB (at least not in this scenario). Probably because VB doesn't allow the user to create a method called "NameOf".

**Comment by thomaslevesque** [comment]
If you can find a scenario where the bug happens in VB, I'll gladly add a test ;)

**Comment by blairconrad** [comment]
> If you can find a scenario where the bug happens in VB, I'll gladly add a test ;)

I'm not that interested. The new code looks good, and I believe it would work even if we were analyzing VB. Diminishing returns. If you've no objections, I'll merge. And I suppose we should release 4.5.1.

**Comment by thomaslevesque** [comment]
> If you've no objections, I'll merge

No objections, go ahead!

**Comment by blairconrad** [comment]
Thanks, @thomaslevesque.

**Pull request merged**